karabo_data/geometry2.py

```diff
@@ -116,7 +243,18 @@ def from_quad_positions(cls, quad_pos, asic_gap=2, panel_gap=29):
         The quadrant positions are given in pixel units, referring to the first
         pixel of the first module in each quadrant, corresponding to data
         channels 0, 4, 8 and 12.
+
+        The origin of the coordinates is in the centre of the detector.
+        Coordinates increase upwards and to the left (looking along the beam).
+
+        To give positions in different units, pass the *unit* parameter as the
+        length of the unit in metres. E.g. ``unit=1e-3`` means the coordinates
+        are in millimetres.
         """
+        px_conversion = unit / cls.pixel_size
```

Comment:
This unit doesn't describe the pixel size (which is fixed for a given detector), but rather what units the quadrant positions (and `asic_gap` and `panel_gap`) are given in. By default they're in pixel units, i.e. 1 means 0.2mm, the side length of 1 pixel, thus `unit=2e-4`. If you prepare the positions in mm, you would use `unit=1e-3` instead.

This might not be used much, but I added it so that the constructors for AGIPD and LPD are a bit more similar, with different units by default instead of hardcoded different units.

I'm adjusting the wording slightly, I hope it will make it clearer.